general/bundler_imports.md

```diff
@@ -0,0 +1,160 @@
+# Bundler Imports
+
+This document is aimed at explaining how npm bundler imports work.
+
+## Why do we need npm bundler imports
+
+As all of you probably know, usually JavaScript web applications need a bundling phase to convert npm modules into a single `.js` file that can be loaded and executed by browsers. The reference tool for this task is [webpack](https://webpack.js.org/).
+
+Sadly, in our DXP platform, we cannot simply create bundles for every portlet and deploy them (or to be more specific: we could, but it would be far from optimal). The main reason being that because our HTML pages are assembled dinamically during runtime (due to portlet layouts being totally configurable), different portlets may need to share the same JavaScript modules.
+
+So, imagine you deploy two portlets (`A` and `B`) and both use `react`. You could create a bundle for each with `webpack`, but that would put two different copies of `react` in the browser memory and you would spend double the bandwidth and processing time fetching and parsing them.
+
+The ideal solution would be making both portlets share the same `react` copy. This way not only you optimize bandwidth, CPU and RAM, but also make sure that both portlets call the same `react` thus minimizing the possibility of issues arising when mixing different versions of the same framework in one single HTML page.
+
+So, how do you do that? That's when npm bundler imports come to the rescue!
```

Comment:
style thing: when → where (sounds more idiomatic to me, but it's also possible the English speaking parts of my brain are damaged now)

Reply:
I initially wrote `where` too, but then changed my mind, don't know why... :-(